x509/extended_key_usage_gen.go

```diff
@@ -145,7 +160,14 @@ func generateIntegerConstants(oidToName map[string]OID) []byte {
 
 func generateNameConstants(oidToName map[string]OID) []byte {
 	buffer := bytes.Buffer{}
-	for _, oid := range oidToName {
+	// Create sorted slice of keys to ensure deterministic output
+	var keys []string
```

Comment:
This is probably better off as an array than a slice so that it doesn't keep getting expanded.

Reply:
The size of `oidToName` isn't known at compile time, so I don't think I could use an array - what about slice with an initial capacity big enough so that no slice expansion is required? Something like `var keys = make([]string, 0, len(oidToName))`